ui/src/logs/utils/matchSections.ts

```diff
@@ -0,0 +1,52 @@
+import {MatchType, Filter, MatchSection} from 'src/types/logs'
+import uuid from 'uuid'
+
+export const getMatchSections = (
+  filters: Filter[],
+  text: string
+): MatchSection[] => {
+  if (filters.length === 0) {
+    return [createSection(MatchType.NONE, text)]
+  }
+
+  try {
+    const pattern = filtersToPattern(filters)
+    return sectionOnPattern(pattern, text)
+  } catch (e) {
+    console.error('Syntax Error: bad search filter expression')
```

Comment:
is it worth presenting this in a user-facing notification by throwing it and consuming it with a `dispatch` of `notify`, or showing something else in the UI?

Reply:
I think I'll refactor here to just pass in a string representing the pattern.